File: src/Domain/Entities/Product.cs
using ProjectTemplate.Domain.Entities;

namespace ProjectTemplate.Domain.Entities;

/// <summary>
/// Product entity example
/// </summary>
public class Product : EntityBase
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public int Stock { get; set; }
    public string Category { get; set; } = string.Empty;
    public new bool IsActive { get; set; } = true;  // Override EntityBase.IsActive
}

File: src/UI/Web/Vue/src/types/product.ts
export interface ProductResponse {
  id: string | number;
  name: string;
  category: string;
  price: number;
  stock: number;
  isActive: boolean;
}

export interface CreateProductRequest {
  name: string;
  category: string;
  price: number;
  stock: number;
  isActive: boolean;
}
